### ISSUE TITLE
Transceiver

### ISSUE THREAD

**Issue by OLLYDOTDEV** [OLLYDOTDEV/Project-Birdseye-DTX-2020#16]
As a User 
I will be needing to have the capability to be used remotely way from my computer
As it much increases the versatility/flexibility and allows a more preventive approach of privacy.

**Comment by OLLYDOTDEV** [comment]
Wireless communication is one of the many things that makes the whole project posable. Having a ROMS is a non-local location that allows ROMS to be Preemptive in the calculation to the time till the "threat" becomes an issue.

Wireless communication is all based on EMF waves but this isn't physics so I won't go into more depth on how they work but rather cover what option I have for Wireless communication.

**Comment by OLLYDOTDEV** [comment]
the classic 433 transmitter and receiver.
Pro's
- cheep 
- has high penetration due to being a reality low frequency
- 

Con's
- low bandwidth
- short-range

**Comment by OLLYDOTDEV** [comment]
NRF24

Pro's


Con's

**Comment by OLLYDOTDEV** [comment]
HC12

Pro's
- works with 3.2 - 5.5v power
- uses serial communication to talk with the controller so nice and easy to use 
- no extra library needed to use 

Con's
-  You have to use 'AT' commands  to control

**Comment by OLLYDOTDEV** [comment]
LoRa

Pro's
- robust 
- low power
- super long-range (6Km)
- uses duel band of 443Mhz or also 1960Mhz
-  Is a newer type of hardware
Con's
- Uses SPA commutation but needs a total of 11 pins to use in total 
- uses 3.3v
- needs a library to work

**Comment by OLLYDOTDEV** [comment]
HC06 Bluetooth module   

Pro's
-  use just to pins to control 
- works with both 3.3v and 5v
- transceiver
- can be connected with smartphones

Con's
-  range about 10m
-  "At" commands to configure

**Comment by OLLYDOTDEV** [comment]
So these are the some RF radios that are used. So to work out what one I am going to use I need to look and see if there is any large issues with any that straight out rules them out from being considered.

classic 433mhz RF transmitters and receiver is the first module that I can see that could be an issue.

1.   the fact that they have not a transceiver meaning that I would need to have two pairs of modules

**Comment by OLLYDOTDEV** [comment]
So these are the some RF radios that are used. So to work out what one I am going to use I need to look and see if there is any large issues with any that straight out rules them out from being considered.

classic 433mhz RF transmitters and receiver is the first module that I can see that could be an issue.

1.   the fact that they have not a transceiver meaning that I would need to have two pairs of modules and that would start to take up a lot of internal room thus taking if off my list 


HC06 Bluetooth module

All in all a great module but Bluetooth

1. It can lose connection in certain conditions like walls as it tends to use low wattage antenna.
2. It has low bandwidth as compared to Wi-Fi.
3. It allows only short-range communication between devices.
4. Security is weak as it can be hacked with ease.

LoRa

while it has to be one of the promising and interesting to knowledge is out there but due to
1. there is the issue that I have have the amount of free pins free thus ruling out this option

**Comment by OLLYDOTDEV** [comment]
I have chosen to use the NRF24 as it just seems to fit my needs best and also what I had accessible on the cheap. so the first issue I needed to fix was getting a 3.3 power #11 

with this, I then tested the theory of the voltage divider on a breadboard.
is I didn't have the right Resistors to I had to add some in serials that way the total Ohms was added.

![IMG_20200729_205207](https://user-images.githubusercontent.com/43589419/88867160-c1e62400-d260-11ea-8c45-108b669e5a7e.jpg)
 
In short, I have a 5v input from USB that then this volage goes orange oval we can se a 2.2 Ohms Resistor

**Comment by OLLYDOTDEV** [comment]
next is to work out what pins I will need to uses that way I can then added it the ROMS schematic for the PCB


https://www.instructables.com/id/NRF24L01-With-ATtiny85-3-Pins/

https://components101.com/wireless/nrf24l01-pinout-features-datasheet

https://howtomechatronics.com/tutorials/arduino/arduino-wireless-communication-nrf24l01-tutorial/

these are the places for my guide to hooking up the NRF24

**Comment by OLLYDOTDEV** [comment]
now while I do have the NRF24 connected to the SPI interface and while I can just upload the .INO file from https://github.com/nRF24/RF24

but when it comes to using this library on the raspberry pi zero w I was having issues 
![image](https://user-images.githubusercontent.com/43589419/89361302-171dac00-d71f-11ea-9230-c08a9b2e5035.png)


thus putting this sprint 5 to a halt as so further continue the user need the raspberry pi zero w set up so that they are able to have a remote control for the ROMS

**Comment by OLLYDOTDEV** [comment]
to See more of the software look at #17

**Comment by OLLYDOTDEV** [comment]
well while I was going to hookuping  the NRF24 to the raspberry pi I worked out that I needed some way to connected so for that i will be on #25. while working i had some large let backs

**Comment by OLLYDOTDEV** [comment]
there is the pinout for the NRF24 is
![image](https://user-images.githubusercontent.com/43589419/90327745-f55bd900-dfea-11ea-9977-306c355cac8f.png)
and there is the pin out of the RPI zero (the w version has the same layout.)
![image](https://user-images.githubusercontent.com/43589419/90327875-e6295b00-dfeb-11ea-9fd2-f834dea37616.png)

so all that I need to do is connect them up and together via the dragram shown below from [tmrh20](http://tmrh20.github.io/RF24/RPi.html).
![image](https://user-images.githubusercontent.com/43589419/90327898-09eca100-dfec-11ea-81ee-6f2e6787570a.png)




with that done I am left with 

![image](https://user-images.githubusercontent.com/43589419/90327979-c7779400-dfec-11ea-9492-0fe176675763.png)

**Comment by OLLYDOTDEV** [comment]
with this the hardware side of the Transceiver is complete
